cowprotocol/cowswap
# Add feature flag for Partial fills using Launch Darkly

# Summary

Supersedes https://github.com/cowprotocol/cowswap/pull/2146

Using Launch Darkly, controls the Partial fills feature

It has 5 distinct envs configured, which should have the equivalent env vars setup on github

- [x] TODO: Add settings to Github secrets 

![image](https://user-images.githubusercontent.com/43217/231516361-2c1c1381-b61c-457a-8e28-ae251972a217.png)

Using the setting named `partialFillsEnabled`, it's possible to toggle it on/off as desired.
Login credentials are on 1password

![image](https://user-images.githubusercontent.com/43217/231514101-1f8bda08-ca40-44cb-9570-0848f9cd60e5.png)

# To Test

1. On Launch darkly panel, make sure the `Test` environment is currently selected and the `partialFillsEnabled` feature flag is `on`
![image](https://user-images.githubusercontent.com/43217/231514470-133a65f7-ffae-4e9b-9065-41e8c428da67.png)
2. Open the limit orders page
* Settings should allow you to toggle partial fills on and off
* Expert mode or regular mode review should allow you to pick an order type different from the settings
3. Place order
* You should be able to place order according to the selection
4. On Launch darkly panel, turn `off` the `partialFillsEnabled` feature flag for the `Test` environment
5. Go back to limit orders page
⚠️ It might take a few minutes for the setting to update. Refreshing forces a re-fetch.
* Settings to change order type should be disabled end default to off
![image](https://user-images.githubusercontent.com/43217/231516984-8ae9cae2-3fd2-4c5f-9bf9-215e165787c7.png)
* Expert mode or regular mode review should NOT allow you to pick an order type and should be set to `fill or kill`
![image](https://user-images.githubusercontent.com/43217/231517061-2ab84920-5c9b-4dee-a68e-5c42aa4f5fd2.png)
6. Place order
* Type should be `fill or kill`
